### PR TITLE
Handle all types of messages

### PR DESCRIPTION
**Pull request by DebXWoody**
Issue: #1366

### CODE REVIEW

**Comment by jubalh** [comment]
Are you sure all the return 1 are good? I'm not sure whether some cases actually depend on the next.

**Comment by DebXWoody** [comment]
Reset and removed the return 1 code. Created new (first) commit.
I need to check some details.

File: src/xmpp/message.c [hunk 1]
```diff
@@ -166,7 +171,11 @@ _message_handler(xmpp_conn_t *const conn, xmpp_stanza_t *const stanza, void *con
         }
     }
 
-    _handle_chat(stanza, FALSE);
+    if (g_strcmp0(type, STANZA_TYPE_CHAT) == 0)  {
```
Comment (jubalh) [inline src/xmpp/message.c:174]: Not sure whether all other messages have type chat.
Maybe it's better to use 
```
if (g_strcmp0(type, "normal") == 0 || type == NULL  )  {
    _handle_normal(stanza);
} else {
  _handle_chat(stanza, FALSE);
}
?

**Comment by jubalh** [comment]
I took the _handle_headline() part and merged it in 3640900cf64f00591526444a80d9c412ba302a0a.
8d8ef675cd6e8d8ee756d247551d255edcea2a87 should then fix a memleak.

I omitted the handle_normal as I'm not sure that's right.

But I'll revisit that during message.c cleanup anyways. (Once I read all XEPs and understand what will be type normal and what will be type chat. It seems in practice its sometimes different that what is said in the XEP).

**Pull request closed without merging**